source/jobs/JobsFeature.cpp

```diff
@@ -180,6 +195,14 @@ void JobsFeature::subscribeToUpdateJobExecutionStatusRejected(string jobId)
         std::bind(
             &JobsFeature::updateJobExecutionStatusRejectedHandler, this, std::placeholders::_1, std::placeholders::_2),
         std::bind(&JobsFeature::ackSubscribeToUpdateJobExecutionRejected, this, std::placeholders::_1));
+
+    if (std::future_status::timeout == updateRejectedPromise.get_future().wait_for(std::chrono::seconds(10)))
+    {
+        ostringstream errorMessage;
+        errorMessage << "Failed to subscribe to updateJobExecutionStatusRejected!";
```

Comment:
Update the error messages everywhere. We should have timed out message here and failure message in respective call back methods.

Reply:
Agree, will update